test/test_nn.py

```diff
@@ -9721,6 +9721,42 @@ def _test_dropout(self, cls, device, input, memory_format=torch.contiguous_forma
         module.__repr__()
         str(module)
 
+    def _test_dropout_discontiguous(self, cls, device, memory_format=torch.contiguous_format):
+        # In this test, we verify that dropout preserves the layout and data for different memory formats.
+        # We check whether, we get same values for the output of dropout, when the probability
+        # of dropout is 0 or very close to 0.
+        # Reference: https://github.com/pytorch/pytorch/issues/47176
+        close_to_zero_p = 0.0000000001  # Should be almost zero but not zero, as for p=0 different path is taken
+        for p in [0, close_to_zero_p]:
+            inp = torch.ones(2, 3, 3, 3, device=device)
+            inp_discontiguous = torch.empty(2, 3, 3, 6, device=device, memory_format=memory_format)[..., ::2]
+            inp_discontiguous.copy_(inp)
+            mod = cls(p=p)
+            out = mod(inp_discontiguous)
+            self.assertEqual(out.layout, inp_discontiguous.layout)
+            self.assertEqual(inp_discontiguous, out)
+
+    def _test_dropout_mean_check(self, cls, device):
+        def _test_dropout_mean(inp, out):
+            self.assertEqual(inp.layout, out.layout)
+            self.assertNotEqual(inp, out)
+            self.assertEqual(inp.mean(), out.mean(), rtol=0.5, atol=0.5)
+
+        def invert_perm(p):
+            d = {x: i for i, x in enumerate(p)}
+            return (d[0], d[1], d[2], d[3])
+
+        inp = torch.ones(2, 3, 4, 5, device=device)
+        shifts = [(0, 0), (1, 0), (0, 1), (1, 1)]
+        for perm in itertools.permutations((0, 1, 2, 3), r=4):
+            for shift in shifts:
+                for p in [0.3, 0.5, 0.7]:
```

Comment:
Can you please add small probability here, and check for that case that all the values are preserved, similar to _test_dropout_discontiguous?